src/univariate/continuous/normalinversegaussian.jl

```diff
@@ -58,3 +58,24 @@ function logpdf(d::NormalInverseGaussian, x::Real)
     μ, α, β, δ = params(d)
     log(α*δ) + log(besselk(1, α*sqrt(δ^2+(x-μ)^2))) - log(π*sqrt(δ^2+(x-μ)^2)) + δ*sqrt(α^2-β^2) + β*(x-μ)
 end
+
+
+#### Sampling
+
+# The Normal Inverse Gaussian distribution is a normal variance-mean
+# mixture with an inverse Gaussian as mixing distribution.
+#
+# Ole E. Barndorff-Nielsen (1997)
+# Normal Inverse Gaussian Distributions and Stochastic Volatility Modelling
+# Scandinavian Journal of Statistics, Vol. 24, pp. 1--13
+# DOI: http://dx.doi.org/10.1111/1467-9469.00045
+
+function rand(rng::Random.AbstractRNG, d::NormalInverseGaussian)
+    μ, α, β, δ = params(d)
+    γ = sqrt(α^2 - β^2)
+
+    Z = InverseGaussian(δ/γ, δ^2)
```

Comment:
From https://en.wikipedia.org/wiki/Inverse_Gaussian_distribution#Scaling and the fact that delta is a scale parameter for the NIG I would almost think `δ` shouldn't be linear in the first and quadratic in the second argument  here.

Reply:
indeed that's not what I read on the paper